spec/ddtrace/contrib/http/request_spec.rb

```diff
@@ -334,4 +334,82 @@ def expect_request_without_distributed_headers
       end
     end
   end
+
+  describe 'request exceptions' do
+    subject(:response) do
+      begin
+        client.get(path)
+      # rubocop:disable Lint/UselessAssignment
+      rescue => e
+        nil
+      end
+    end
+
+    let(:path) { '/my/path' }
+
+    context 'that raises a timeout exception' do
+      let(:timeout_error) { Net::OpenTimeout.new }
+
+      before(:each) { stub_request(:any, "#{uri}#{path}").to_timeout }
+
+      let(:span) { spans.first }
+
+      it 'generates a well-formed trace with span tags available from request object' do
+        expect(response).to eq(nil)
+        expect(spans).to have(1).items
+        expect(span.name).to eq('http.request')
+        expect(span.service).to eq('net/http')
+        expect(span.resource).to eq('GET')
+        expect(span.get_tag('http.url')).to eq(path)
+        expect(span.get_tag('http.method')).to eq('GET')
+        expect(span.get_tag('out.host')).to eq(host)
+        expect(span.get_tag('out.port')).to eq(port.to_s)
+        expect(span.status).to eq(1)
```

Comment:
I would recommend using the helpers @marcotc introduced here instead of reading the tags directly, e.g. https://github.com/DataDog/dd-trace-rb/blob/master/spec/ddtrace/contrib/faraday/middleware_spec.rb#L112-L114

Reply:
good to know those are there, updated.